### PR TITLE
feat(*): added scale radial

### PR DESCRIPTION
**Pull request by sarathps93**
#### :rocket: Enhancements

- Added `Radial Scale` which will help in creating charts like `circular bar plot`. Adding it as a part of #832

### CODE REVIEW

**Comment by sarathps93** [comment]
Had to update the version of `d3-scale` and it's types package to get the `scaleRadial` and type from it

File: packages/visx-scale/src/operators/unknown.ts [hunk 1]
```diff
@@ -11,6 +11,6 @@ export default function applyUnknown<
   config: ScaleConfigWithoutType<Output, DiscreteInput, ThresholdInput>,
 ) {
   if ('unknown' in scale && 'unknown' in config && typeof config.unknown !== 'undefined') {
-    scale.unknown(config.unknown);
+    (scale.unknown as any)(config.unknown);
```
Comment (sarathps93) [inline packages/visx-scale/src/operators/unknown.ts:14]: The latest type definitions for `d3-scales` are causing issues at here and that's why I have including the type casting in here

File: packages/visx-scale/src/types/ScaleConfig.ts [hunk 2]
```diff
@@ -55,6 +55,13 @@ export type SymlogScaleConfig<Output = DefaultOutput> = CreateScaleConfig<
   'clamp' | 'constant' | 'nice' | 'zero'
 >;
 
+export type RadialScaleConfig<Output = DefaultOutput> = CreateScaleConfig<
+  'radial',
+  ContinuousDomain,
+  Output[],
+  'clamp' | 'nice' | 'round' | 'zero' | 'unknown'
```
Comment (williaster) [inline packages/visx-scale/src/types/ScaleConfig.ts:62]: should `zero` be omitted here since it's not applied in `scales/radial.ts`?

**Review by williaster — COMMENTED** [review]
This looks great to me with one minor comment on the `RadialScaleConfig`.

Also cc-ing @kristw in case he notices anything else since he's very familiar with the scale package.

File: packages/visx-scale/src/types/ScaleConfig.ts (hunk 2 above)
Comment (sarathps93) [reply packages/visx-scale/src/types/ScaleConfig.ts]: Yeah makes sense. Have done the changes

Comment (kristw) [reply packages/visx-scale/src/types/ScaleConfig.ts]: +1

File: packages/visx-scale/src/operators/unknown.ts (hunk 1 above)
Comment (kristw) [reply packages/visx-scale/src/operators/unknown.ts:14]: Would you mind sharing the specific issue? 
Just want to help think if there is a workaround that is more type-safe than casting to `any`.

Comment (sarathps93) [reply packages/visx-scale/src/operators/unknown.ts:14]: ` /**
     * Returns the current unknown value, which defaults to undefined.
     */
    unknown(): UnknownReturnType<Unknown, undefined>;
    /**
     * Sets the output value of the scale for undefined (or NaN) input values and returns this scale.
     *
     * @param value The output value of the scale for undefined (or NaN) input values.
     */
    unknown<NewUnknown>(value: NewUnknown): ScaleLinear<Range, Output, NewUnknown>;`

Above is the latest type definition from `@types/d3-scale`. Somehow the `scale.unknown` is pointing to the function which accepts no arguments.

Comment (sarathps93) [reply packages/visx-scale/src/operators/unknown.ts:14]: @kristw Please let me if any other workaround can be done to avoid this.

Comment (williaster) [reply packages/visx-scale/src/operators/unknown.ts:14]: It looks like TS isn't correctly matching the second overload. is there any way to import `NewUnknown` and coerce to that?

`scale.unknown(config.unknown as NewUnknown)`

Comment (sarathps93) [reply packages/visx-scale/src/operators/unknown.ts:14]: @williaster I have tried that option as well. They are not exposing the `NewUnknown` for us to import

Comment (williaster) [reply packages/visx-scale/src/operators/unknown.ts:14]: @kristw any other thoughts? else we can probably merge this as is

**Review by williaster — APPROVED** [review]
Spoke with @kristw offline, let's merge this! Thanks again for the contribution 🙏

**Pull request merged**